src/registry/selectboxes/index.tsx

```diff
@@ -50,6 +50,20 @@ export const FormioSelectboxes: React.FC<FormioSelectboxesProps> = ({componentDe
 
     const {value} = getFieldProps(key);
 
+    // Disable rest of the checkboxes if maxSelectedCount has been reached
+    if (maxSelectedCount) {
```

Comment:
this is a good example of where synchronizing state with an effect is an anti pattern - we already have existing state/information to figure out if/what needs to be disabled, so the entire additional `useState` is not necessary.

I've refactored this, take a look so you see what I mean :-)